File: app/src/googleplay/java/org/tasks/billing/SignatureVerifier.java
package org.tasks.billing;

import android.content.Context;
import java.io.IOException;
import javax.inject.Inject;
import org.tasks.R;
import org.tasks.injection.ForApplication;
import timber.log.Timber;

class SignatureVerifier {

  private final String billingKey;

  @Inject
  SignatureVerifier(@ForApplication Context context) {
    billingKey = context.getString(R.string.gp_key);
  }

  boolean verifySignature(Purchase purchase) {
    try {
      return Security.verifyPurchase(
          billingKey, purchase.getOriginalJson(), purchase.getSignature());
    } catch (IOException e) {
      Timber.e(e);
      return false;
    }
  }
}

File: app/src/googleplay/java/org/tasks/billing/SignatureVerifier.kt
package org.tasks.billing

import android.content.Context
import dagger.hilt.android.qualifiers.ApplicationContext
import org.tasks.R
import timber.log.Timber
import java.io.IOException
import javax.inject.Inject

class SignatureVerifier @Inject constructor(@ApplicationContext context: Context) {
    private val billingKey: String = context.getString(R.string.gp_key)

    fun verifySignature(purchase: Purchase): Boolean {
        return try {
            Security.verifyPurchase(
                    billingKey, purchase.originalJson, purchase.signature)
        } catch (e: IOException) {
            Timber.e(e)
            false
        }
    }
}